src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractTypeImportCompletionService.CacheEntry.cs

```diff
@@ -25,14 +25,18 @@ internal abstract partial class AbstractTypeImportCompletionService
 
             private ImmutableArray<TypeImportCompletionItemInfo> ItemInfos { get; }
 
+            public bool ContainsAdvancedMembers { get; }
```

Comment:
it's unclear if we actually need this member.  is it just a perf optimization?  given that C# will almost always be on 'show advanced members', i don't think this buys us much.

Reply:
Yeah, when 'show advanced members` is on, this doesn't do anything. But on the other hand, having this cost us nothing (in terms of perf and code complexity) but would improve perf when 'show advanced members` is off. I don't really see a downside to this.

Reply:
well, it made the code more confusing for me :)   so that was one downside :)

Reply:
OK, let me add a few comments then. The extra code related to this really is just 2 conditionals.